### PR TITLE
feat: custom directories

### PR DESCRIPTION
**Pull request by ricardogobbosouza**
Feature custom directories in `nuxt.config.js`.

```js
module.exports = {
  dir: {
    assets: 'custom-assets-dir', // Default: 'assets'
    layouts: 'custom-layouts-dir', // Default: 'layouts'
    middleware: 'custom-middleware-dir', // Default: 'middleware'
    pages: 'custom-pages-dir', // Default: 'pages'
    static: 'custom-static-dir', // Default: 'static'
    store: 'custom-store-dir' // Default: 'store'
  }
}
```

### CODE REVIEW

File: lib/common/options.js [hunk 1]
```diff
@@ -284,6 +284,9 @@ Options.defaults = {
     name: 'layout',
     mode: 'out-in'
   },
+  assetsDir: 'assets',
```
Comment (pi0) [inline lib/common/options.js]: If it is going to be be a best practice, i think we may move all of this options to a new `dir` block:
```js
dir: {
  assets: ...
}
```

Comment (ricardogobbosouza) [reply lib/common/options.js]: Yes, you're right, I'm moving.

**Comment by frizadiga** [comment]
I think its a good idea to give developer flexibility to changed dir name or path such as `layouts, static, store, pages` in the nuxt.config.js, hope this PR will be merged as soon as possible.

**Comment by Atinux** [comment]
Thank you for this PR @frizadiga

**Pull request merged**